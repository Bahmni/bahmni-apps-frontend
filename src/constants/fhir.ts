export const FHIR_ENCOUNTER_TYPE_CODE_SYSTEM =
  'http://fhir.openmrs.org/code-system/encounter-type';
export const HL7_CONDITION_CATEGORY_CONDITION_CODE = 'problem-list-item';
