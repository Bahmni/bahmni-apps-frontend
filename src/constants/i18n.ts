export const LOCALE_COOKIE_NAME = "NG_TRANSLATE_LANG_KEY";