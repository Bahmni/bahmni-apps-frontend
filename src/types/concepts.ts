export interface ConceptSearch {
  conceptName: string;
  conceptUuid: string;
  matchedName: string;
  disabled?: boolean;
}
