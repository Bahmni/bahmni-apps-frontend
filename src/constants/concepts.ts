import { Coding } from 'fhir/r4';
import { AllergenType } from '@types/concepts';

export const CERTAINITY_CONCEPTS: Coding[] = [
  {
    code: 'confirmed',
    display: 'CERTAINITY_CONFIRMED',
    system: 'http://terminology.hl7.org/CodeSystem/condition-ver-status',
  },
  {
    code: 'provisional',
    display: 'CERTAINITY_PROVISIONAL',
    system: 'http://terminology.hl7.org/CodeSystem/condition-ver-status',
  },
];

export const ALLERGEN_TYPES: Record<
  Uppercase<AllergenType>,
  {
    code: string;
    display: AllergenType;
    system: string;
  }
> = {
  FOOD: {
    code: '162553AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
    display: 'food',
    system: 'http://terminology.hl7.org/CodeSystem/allergyintolerance-type',
  },
  MEDICATION: {
    code: '162552AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
    display: 'medication',
    system: 'http://terminology.hl7.org/CodeSystem/allergyintolerance-type',
  },
  ENVIRONMENT: {
    code: '162554AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
    display: 'environment',
    system: 'http://terminology.hl7.org/CodeSystem/allergyintolerance-type',
  },
};

export const ALLERGY_REACTION: Coding = {
  code: '162555AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
  display: 'reaction',
  system: 'http://terminology.hl7.org/CodeSystem/allergyintolerance-type',
};

export const ALLERGY_SEVERITY = {
  MILD: 'mild',
  MODERATE: 'moderate',
  SEVERE: 'severe',
};
