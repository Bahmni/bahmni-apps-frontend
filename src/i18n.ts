import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import Backend from 'i18next-http-backend';
import LanguageDetector from 'i18next-browser-languagedetector';

const mergeTranslations = async (lng: string) => {
  const standardConfigUrl = `bahmni_config/openmrs/i18n/home/locale_${lng}.json`;
  const localConfigUrl = `/locales/locale_${lng}.json`;
  const [standardConfig, localConfig] = await Promise.all([
    fetch(standardConfigUrl).then((res) => res.json()),
    fetch(localConfigUrl).then((res) => res.json()),
  ]);
  return { ...localConfig,  ...standardConfig }; // Local config overrides standard config
};

i18n
  .use(Backend)
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    fallbackLng: 'en',
    debug: process.env.NODE_ENV === 'development',
    ns: ['translation'],
    defaultNS: process.env.REACT_APP_DEFAULT_NAMESPACE || 'translation',

    detection: {
      order: ['cookie', 'localStorage', 'navigator', 'htmlTag'],
      lookupCookie: 'NG_TRANSLATE_LANG_KEY', // Match Bahmni's existing cookie name
      caches: ['cookie'],
      cookieOptions: {
        path: '/',
        sameSite: 'strict'
      }},
    interpolation: {
      escapeValue: false,
    },
    backend: {
      loadPath: '/locales/locale_{{lng}}.json',
      parse: async (data: string, lng: string) => {
        const mergedTranslations = await mergeTranslations(lng);
        return mergedTranslations;
      }
    }, // Properly closed backend object
    react: {
      useSuspense: true,
    },
  });

export default i18n;
