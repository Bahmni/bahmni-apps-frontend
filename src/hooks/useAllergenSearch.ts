import { useEffect, useState, useMemo } from 'react';
import useDebounce from './useDebounce';
import { fetchAndFormatAllergenConcepts } from '@services/allergenService';
import { AllergenConcept } from '@types/concepts';
import { getFormattedError } from '@utils/common';

interface UseAllergenSearchResult {
  allergens: AllergenConcept[];
  isLoading: boolean;
  error: Error | null;
}

/**
 * A hook that provides debounced search functionality for allergen concepts.
 * It eagerly loads all allergen concepts and filters them based on the search term.
 * @param searchTerm - Optional search term to filter allergens
 * @returns Object containing filtered allergens, loading state, and any errors
 */
const useAllergenSearch = (
  searchTerm: string = '',
): UseAllergenSearchResult => {
  const [allergens, setAllergens] = useState<AllergenConcept[]>([]);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState<Error | null>(null);
  const debouncedSearchTerm = useDebounce(searchTerm);

  // Load all allergens on mount
  useEffect(() => {
    const loadAllergens = async () => {
      setIsLoading(true);
      setError(null);
      try {
        const concepts = await fetchAndFormatAllergenConcepts();
        setAllergens(concepts);
      } catch (err) {
        const formattedError = getFormattedError(err);
        setError(new Error(formattedError.message));
      } finally {
        setIsLoading(false);
      }
    };
    loadAllergens();
  }, []);

  // Filter allergens based on search term
  const filteredAllergens = useMemo(() => {
    if (!allergens.length) return [];

    const searchTermLower = debouncedSearchTerm?.toLowerCase().trim() || '';
    if (!searchTermLower) return allergens;

    return allergens.filter((allergen) => {
      const displayLower = allergen.display.trim().toLowerCase();
      return allergen && displayLower.includes(searchTermLower);
    });
  }, [allergens, debouncedSearchTerm]);

  return {
    allergens: filteredAllergens,
    isLoading,
    error,
  };
};

export default useAllergenSearch;
