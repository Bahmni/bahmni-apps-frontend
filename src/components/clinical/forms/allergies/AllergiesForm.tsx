import { ComboBox, Tile } from '@carbon/react';
import React, { useState, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import BoxWHeader from '@components/common/boxWHeader/BoxWHeader';
import SelectedItem from '@components/common/selectedItem/SelectedItem';
import useAllergenSearch from '@hooks/useAllergenSearch';
import { useAllergyStore } from '@stores/allergyStore';
import { AllergenConcept } from '@types/concepts';
import { getCategoryDisplayName } from '@utils/allergy';
import SelectedAllergyItem from './SelectedAllergyItem';
import * as styles from './styles/AllergiesForm.module.scss';

/**
 * AllergiesForm component
 *
 * A component that displays a search interface for allergies and a list of selected allergies.
 * It allows users to search for allergies, select them, and specify severity and reactions.
 */
const AllergiesForm: React.FC = React.memo(() => {
  const { t } = useTranslation();
  const [searchAllergenTerm, setSearchAllergenTerm] = useState('');

  // Use Zustand store
  const {
    selectedAllergies,
    addAllergy,
    removeAllergy,
    updateSeverity,
    updateReactions,
    updateNote,
  } = useAllergyStore();

  // Use allergen search hook
  const {
    allergens: searchResults,
    reactions: reactionConcepts,
    isLoading,
    error,
  } = useAllergenSearch(searchAllergenTerm);

  const handleSearch = (searchTerm: string) => {
    setSearchAllergenTerm(searchTerm);
  };

  const handleOnChange = (
    selectedItem:
      | AllergenConcept
      | { uuid: string; display: string; type: null; disabled: boolean }
      | null,
  ) => {
    if (
      !selectedItem ||
      !selectedItem.uuid ||
      !selectedItem.display ||
      !selectedItem.type
    ) {
      return;
    }

    addAllergy(selectedItem as AllergenConcept);
  };

  const filteredSearchResults = useMemo(() => {
    if (searchAllergenTerm.length === 0) return [];
    if (isLoading) {
      return [
        {
          uuid: '',
          display: t('LOADING_CONCEPTS'),
          type: null,
          disabled: isLoading,
        },
      ];
    }
    const isSearchEmpty = searchResults.length === 0 && !error;

    if (isSearchEmpty) {
      return [
        {
          uuid: '',
          display: t('NO_MATCHING_ALLERGEN_FOUND'),
          type: null,
          disabled: isSearchEmpty,
        },
      ];
    }

    if (error) {
      return [
        {
          uuid: '',
          display: t('ERROR_FETCHING_CONCEPTS'),
          type: null,
          disabled: true,
        },
      ];
    }

    return searchResults.map((item) => {
      const isAlreadySelected = selectedAllergies.some(
        (a) => a.id === item.uuid,
      );
      return {
        ...item,
        display: isAlreadySelected
          ? `${item.display} (${t('ALLERGY_ALREADY_SELECTED')})`
          : item.display,
        type: isAlreadySelected ? null : item.type,
        disabled: isAlreadySelected,
      };
    });
  }, [
    isLoading,
    searchResults,
    searchAllergenTerm,
    error,
    selectedAllergies,
    t,
  ]);

  return (
    <Tile className={styles.allergiesFormTile}>
      <div className={styles.allergiesFormTitle}>
        {t('ALLERGIES_FORM_TITLE')}
      </div>
      <ComboBox
        id="allergies-search"
        placeholder={t('ALLERGIES_SEARCH_PLACEHOLDER')}
        items={filteredSearchResults}
        itemToString={(item) =>
          item?.type
            ? `${item.display} [${t(getCategoryDisplayName(item.type))}]`
            : item
              ? `${item.display}`
              : ''
        }
        onChange={(data) => handleOnChange(data.selectedItem!)}
        onInputChange={(searchQuery: string) => handleSearch(searchQuery)}
        size="md"
        autoAlign
        aria-label={t('ALLERGIES_SEARCH_ARIA_LABEL')}
      />
      {selectedAllergies && selectedAllergies.length > 0 && (
        <BoxWHeader
          title={t('ALLERGIES_ADDED_ALLERGIES')}
          className={styles.allergiesBox}
        >
          {selectedAllergies.map((allergy) => (
            <SelectedItem
              key={allergy.id}
              className={styles.selectedAllergyItem}
              onClose={() => removeAllergy(allergy.id)}
            >
              <SelectedAllergyItem
                allergy={allergy}
                reactionConcepts={reactionConcepts}
                updateSeverity={updateSeverity}
                updateReactions={updateReactions}
                updateNote={updateNote}
              />
            </SelectedItem>
          ))}
        </BoxWHeader>
      )}
    </Tile>
  );
});

AllergiesForm.displayName = 'AllergiesForm';

export default AllergiesForm;
