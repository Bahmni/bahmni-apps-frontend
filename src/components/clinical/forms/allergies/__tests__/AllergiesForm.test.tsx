import React from 'react';
import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { I18nextProvider } from 'react-i18next';
import i18n from '@/setupTests.i18n';
import AllergiesForm from '../AllergiesForm';
import { useAllergyStore } from '@stores/allergyStore';
import useAllergenSearch from '@hooks/useAllergenSearch';
import { axe, toHaveNoViolations } from 'jest-axe';
import { AllergenConcept } from '@types/concepts';
import { Coding } from 'fhir/r4';
expect.extend(toHaveNoViolations);

// Mock the store and hooks
jest.mock('@stores/allergyStore');
jest.mock('@hooks/useAllergenSearch');

// Mock CSS modules
jest.mock('../styles/AllergiesForm.module.scss', () => ({
  allergiesFormTile: 'allergiesFormTile',
  allergiesFormTitle: 'allergiesFormTitle',
  allergiesBox: 'allergiesBox',
  selectedAllergyItem: 'selectedAllergyItem',
}));

const mockAllergen: AllergenConcept = {
  uuid: 'test-allergy-1',
  display: 'Peanut Allergy',
  type: 'food',
  disabled: false,
};

const mockReactions: Coding[] = [
  {
    code: 'hives',
    display: 'REACTION_HIVES',
    system: 'http://snomed.info/sct',
  },
  {
    code: 'rash',
    display: 'REACTION_RASH',
    system: 'http://snomed.info/sct',
  },
];

const mockStore = {
  selectedAllergies: [],
  addAllergy: jest.fn(),
  removeAllergy: jest.fn(),
  updateSeverity: jest.fn(),
  updateReactions: jest.fn(),
  validateAllAllergies: jest.fn(),
  reset: jest.fn(),
  getState: jest.fn(),
};

const mockAllergenSearch = {
  allergens: [],
  reactions: mockReactions,
  isLoading: false,
  error: null,
};

describe('AllergiesForm', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.spyOn(console, 'error').mockImplementation(() => {});
    window.HTMLElement.prototype.scrollIntoView = jest.fn();
    (useAllergyStore as unknown as jest.Mock).mockReturnValue(mockStore);
    (useAllergenSearch as jest.Mock).mockReturnValue(mockAllergenSearch);
    i18n.changeLanguage('en');
  });

  // HAPPY PATH TESTS
  describe('Happy Path Scenarios', () => {
    test('renders search box correctly', () => {
      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      expect(
        screen.getByRole('combobox', { name: /search for allergies/i }),
      ).toBeInTheDocument();
    });

    test('adds an allergy when selected from search', async () => {
      const user = userEvent.setup();
      (useAllergenSearch as jest.Mock).mockReturnValue({
        ...mockAllergenSearch,
        allergens: [mockAllergen],
      });

      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      const searchBox = screen.getByRole('combobox', {
        name: /search for allergies/i,
      });

      // Type in the search box
      await waitFor(async () => {
        await user.type(searchBox, 'peanut');
      });

      // Wait for the dropdown item to appear
      await waitFor(() => {
        expect(screen.getByText('Peanut Allergy [Food]')).toBeInTheDocument();
      });

      // Click on the dropdown item
      await waitFor(async () => {
        await user.click(screen.getByText('Peanut Allergy [Food]'));
      });

      // Verify the store was called correctly
      await waitFor(() => {
        expect(mockStore.addAllergy).toHaveBeenCalledWith(mockAllergen);
      });
    });

    test('displays selected allergies', () => {
      (useAllergyStore as unknown as jest.Mock).mockReturnValue({
        ...mockStore,
        selectedAllergies: [
          {
            id: mockAllergen.uuid,
            display: mockAllergen.display,
            selectedSeverity: null,
            selectedReactions: [],
            errors: {},
            hasBeenValidated: false,
          },
        ],
      });

      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      expect(screen.getByText(/Peanut Allergy/)).toBeInTheDocument();
    });

    test('removes an allergy when close button is clicked', async () => {
      const user = userEvent.setup();
      (useAllergyStore as unknown as jest.Mock).mockReturnValue({
        ...mockStore,
        selectedAllergies: [
          {
            id: mockAllergen.uuid,
            display: mockAllergen.display,
            selectedReactions: [],
            errors: {},
            hasBeenValidated: false,
          },
        ],
      });

      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      // Use data-testid to find the close button
      const closeButton = screen.getByTestId('selected-item-close-button');
      await waitFor(async () => {
        await user.click(closeButton);
      });

      await waitFor(() => {
        expect(mockStore.removeAllergy).toHaveBeenCalledWith(mockAllergen.uuid);
      });
    });
  });

  // SAD PATH TESTS
  describe('Sad Path Scenarios', () => {
    test('shows loading state while searching', async () => {
      const user = userEvent.setup();
      (useAllergenSearch as jest.Mock).mockReturnValue({
        ...mockAllergenSearch,
        isLoading: true,
      });

      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      // Type something in the search box to trigger the loading state
      const searchBox = screen.getByRole('combobox', {
        name: /search for allergies/i,
      });
      await waitFor(async () => {
        await user.type(searchBox, 'a');
      });
      // Use a more flexible text matcher
      await waitFor(() => {
        expect(screen.getByText(/loading concepts/i)).toBeInTheDocument();
      });
    });

    test('shows error when search fails', async () => {
      const user = userEvent.setup();
      (useAllergenSearch as jest.Mock).mockReturnValue({
        ...mockAllergenSearch,
        error: new Error('Failed to load allergens'),
      });

      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      // Type something in the search box to trigger the search
      const searchBox = screen.getByRole('combobox', {
        name: /search for allergies/i,
      });
      await waitFor(async () => {
        await user.type(searchBox, 'a');
      });

      // Use a more flexible text matcher
      await waitFor(() => {
        expect(
          screen.getByText(/unexpected error occurred/i),
        ).toBeInTheDocument();
      });
    });

    test('shows message when no search results found', async () => {
      const user = userEvent.setup();
      (useAllergenSearch as jest.Mock).mockReturnValue({
        ...mockAllergenSearch,
        allergens: [], // Empty array to simulate no results
      });

      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      // Type something in the search box to trigger the search
      const searchBox = screen.getByRole('combobox', {
        name: /search for allergies/i,
      });

      await waitFor(async () => {
        await user.type(searchBox, 'nonexistent');
      });

      // Use a more flexible text matcher
      await waitFor(() => {
        expect(
          screen.getByText(/No matching allergen recorded/i),
        ).toBeInTheDocument();
      });
    });

    test('does not add allergy when selected item is invalid', async () => {
      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      const searchBox = screen.getByRole('combobox', {
        name: /search for allergies/i,
      });

      // Test with undefined selectedItem
      await waitFor(() => {
        fireEvent.change(searchBox, {
          target: { value: 'test' },
          selectedItem: undefined,
        });
      });
      expect(mockStore.addAllergy).not.toHaveBeenCalled();

      // Test with missing uuid
      await waitFor(() => {
        fireEvent.change(searchBox, {
          target: { value: 'test' },
          selectedItem: { display: 'Test Allergy' },
        });
      });
      expect(mockStore.addAllergy).not.toHaveBeenCalled();

      // Test with missing display
      await waitFor(() => {
        fireEvent.change(searchBox, {
          target: { value: 'test' },
          selectedItem: { uuid: 'test-uuid' },
        });
      });
      expect(mockStore.addAllergy).not.toHaveBeenCalled();
    });

    test('prevents adding duplicate allergies', async () => {
      const user = userEvent.setup();
      (useAllergenSearch as jest.Mock).mockReturnValue({
        ...mockAllergenSearch,
        allergens: [mockAllergen],
      });
      (useAllergyStore as unknown as jest.Mock).mockReturnValue({
        ...mockStore,
        selectedAllergies: [
          {
            id: mockAllergen.uuid,
            display: mockAllergen.display,
            selectedReactions: [],
            errors: {},
            hasBeenValidated: false,
          },
        ],
      });

      render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      const searchBox = screen.getByRole('combobox', {
        name: /search for allergies/i,
      });

      await waitFor(async () => {
        await user.type(searchBox, 'peanut');
      });

      // Use the correct text that appears in the DOM
      await waitFor(async () => {
        await user.click(screen.getByText('Peanut Allergy (Already added)'));
      });

      await waitFor(() => {
        expect(mockStore.addAllergy).not.toHaveBeenCalled();
        // The test is also looking for "Allergy already selected" which might not be correct
        // Let's use a more flexible approach
        expect(screen.getByText(/already added/i)).toBeInTheDocument();
      });
    });
  });

  // ACCESSIBILITY TESTS
  describe('Accessibility', () => {
    test('should have no accessibility violations', async () => {
      const { container } = render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );

      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });

  // SNAPSHOT TESTS
  describe('Snapshot Tests', () => {
    test('matches snapshot with no allergies', () => {
      const { container } = render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );
      expect(container).toMatchSnapshot();
    });

    test('matches snapshot with selected allergies', () => {
      (useAllergyStore as unknown as jest.Mock).mockReturnValue({
        ...mockStore,
        selectedAllergies: [
          {
            id: mockAllergen.uuid,
            display: mockAllergen.display,
            selectedReactions: [mockReactions[0]],
            errors: {},
            hasBeenValidated: true,
          },
        ],
      });

      const { container } = render(
        <I18nextProvider i18n={i18n}>
          <AllergiesForm />
        </I18nextProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
