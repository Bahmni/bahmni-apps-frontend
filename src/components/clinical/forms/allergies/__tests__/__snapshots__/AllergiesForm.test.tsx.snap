// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllergiesForm Snapshot Tests matches snapshot with no allergies 1`] = `
<div>
  <div
    class="cds--tile allergiesFormTile"
  >
    <div
      class="allergiesFormTitle"
    >
      Allergies
    </div>
    <div
      class="cds--list-box__wrapper"
    >
      <div
        class="cds--combo-box cds--autoalign cds--list-box cds--list-box--lg"
      >
        <div
          class="cds--list-box__field"
        >
          <input
            aria-activedescendant=""
            aria-autocomplete="list"
            aria-controls="downshift-«r1u»-menu"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Search for allergies"
            autocomplete="off"
            class="cds--text-input cds--text-input--empty"
            id="allergies-search"
            placeholder="Search to add new allergy"
            role="combobox"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
          <button
            aria-controls="downshift-«r1u»-menu"
            aria-expanded="false"
            aria-label="Open"
            class="cds--list-box__menu-icon"
            id="downshift-«r1u»-toggle-button"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
              />
            </svg>
          </button>
        </div>
        <ul
          aria-labelledby="downshift-«r1u»-label"
          class="cds--list-box__menu"
          id="downshift-«r1u»-menu"
          role="listbox"
          style="position: fixed; left: 0px; top: 0px; visibility: visible; transform: translate(0px, 0px);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`AllergiesForm Snapshot Tests matches snapshot with selected allergies 1`] = `
<div>
  <div
    class="cds--tile allergiesFormTile"
  >
    <div
      class="allergiesFormTitle"
    >
      Allergies
    </div>
    <div
      class="cds--list-box__wrapper"
    >
      <div
        class="cds--combo-box cds--autoalign cds--list-box cds--list-box--lg"
      >
        <div
          class="cds--list-box__field"
        >
          <input
            aria-activedescendant=""
            aria-autocomplete="list"
            aria-controls="downshift-«r21»-menu"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-label="Search for allergies"
            autocomplete="off"
            class="cds--text-input cds--text-input--empty"
            id="allergies-search"
            placeholder="Search to add new allergy"
            role="combobox"
            tabindex="0"
            title=""
            type="text"
            value=""
          />
          <button
            aria-controls="downshift-«r21»-menu"
            aria-expanded="false"
            aria-label="Open"
            class="cds--list-box__menu-icon"
            id="downshift-«r21»-toggle-button"
            tabindex="-1"
            title="Open"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
              />
            </svg>
          </button>
        </div>
        <ul
          aria-labelledby="downshift-«r21»-label"
          class="cds--list-box__menu"
          id="downshift-«r21»-menu"
          role="listbox"
          style="position: fixed; left: 0px; top: 0px; visibility: visible; transform: translate(0px, 0px);"
        />
      </div>
    </div>
    <div
      aria-label="Added Allergies"
      class="allergiesBox cds--css-grid cds--css-grid--narrow cds--css-grid--full-width"
      data-testid="box-w-title"
      role="region"
    >
      <div
        class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-16 cds--xlg:col-span-16 cds--css-grid-column"
      >
        Added Allergies
      </div>
      <div
        class="cds--sm:col-span-4 cds--md:col-span-8 cds--lg:col-span-16 cds--xlg:col-span-16 cds--css-grid-column"
      >
        <div
          class="selectedAllergyItem cds--subgrid cds--subgrid--narrow"
          data-testid="selected-item"
        >
          <div
            class="cds--sm:col-span-3 cds--md:col-span-7 cds--lg:col-span-15 cds--xlg:col-span-15 cds--css-grid-column"
          >
            <div
              class="cds--subgrid cds--subgrid--wide"
            >
              <div
                class="cds--sm:col-span-4 cds--md:col-span-5 cds--lg:col-span-8 cds--xlg:col-span-8 cds--css-grid-column"
              >
                Peanut Allergy
                 [
                undefined
                ]
              </div>
              <div
                class="cds--sm:col-span-4 cds--md:col-span-3 cds--lg:col-span-3 cds--xlg:col-span-3 cds--css-grid-column"
              >
                <div
                  class="cds--dropdown__wrapper cds--list-box__wrapper"
                  data-testid="allergy-severity-dropdown-test-allergy-1"
                >
                  <label
                    class="cds--label cds--visually-hidden"
                    for="downshift-«r24»-toggle-button"
                    id="downshift-«r24»-label"
                  >
                    Select Severity
                  </label>
                  <div
                    class="cds--dropdown cds--autoalign cds--list-box"
                    id="allergy-severity-dropdown-test-allergy-1"
                  >
                    <button
                      aria-activedescendant=""
                      aria-controls="downshift-«r24»-menu"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-label="Select severity for this allergy"
                      class="cds--list-box__field"
                      id="downshift-«r24»-toggle-button"
                      role="combobox"
                      tabindex="0"
                      title="Select Severity"
                      type="button"
                    >
                      <span
                        class="cds--list-box__label"
                      >
                        Select Severity
                      </span>
                      <div
                        class="cds--list-box__menu-icon"
                      >
                        <svg
                          aria-label="Open menu"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          name="chevron--down"
                          preserveAspectRatio="xMidYMid meet"
                          role="img"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                          />
                          <title>
                            Open menu
                          </title>
                        </svg>
                      </div>
                    </button>
                    <ul
                      aria-labelledby="downshift-«r24»-label"
                      class="cds--list-box__menu"
                      id="downshift-«r24»-menu"
                      role="listbox"
                      style="position: fixed; left: 0px; top: 0px; visibility: visible; transform: translate(0px, 0px);"
                    />
                  </div>
                </div>
              </div>
              <div
                class="cds--sm:col-span-4 cds--md:col-span-4 cds--lg:col-span-4 cds--xlg:col-span-4 cds--css-grid-column"
              >
                <div
                  class="cds--multi-select__wrapper cds--multi-select--filterable__wrapper cds--list-box__wrapper cds--autoalign"
                >
                  <label
                    class="cds--label cds--visually-hidden"
                    for="allergy-reactions-multiselect-test-allergy-1-input"
                    id="allergy-reactions-multiselect-test-allergy-1-label"
                  >
                    Select Reactions
                    <span
                      class="cds--visually-hidden"
                    >
                      Total items selected:  1. To clear selection, press Delete or Backspace.
                    </span>
                  </label>
                  <div
                    class="cds--multi-select cds--combo-box cds--multi-select--filterable cds--multi-select--selected cds--list-box"
                    id="allergy-reactions-multiselect-test-allergy-1"
                  >
                    <div
                      class="cds--list-box__field"
                    >
                      <div
                        class="cds--tag cds--tag--filter cds--tag--high-contrast"
                      >
                        <span
                          class="cds--tag__label"
                          title="1"
                        >
                          1
                        </span>
                        <button
                          aria-label="Clear all selected items"
                          class="cds--tag__close-icon"
                          tabindex="-1"
                          title="Clear all selected items"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            focusable="false"
                            height="16"
                            preserveAspectRatio="xMidYMid meet"
                            viewBox="0 0 32 32"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M17.4141 16L24 9.4141 22.5859 8 16 14.5859 9.4143 8 8 9.4141 14.5859 16 8 22.5859 9.4143 24 16 17.4141 22.5859 24 24 22.5859 17.4141 16z"
                            />
                          </svg>
                        </button>
                      </div>
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        autocomplete="off"
                        class="cds--text-input cds--text-input--empty"
                        id="allergy-reactions-multiselect-test-allergy-1-input"
                        placeholder="Select Reactions"
                        role="combobox"
                        value=""
                      />
                      <button
                        aria-controls="allergy-reactions-multiselect-test-allergy-1__menu"
                        aria-expanded="false"
                        aria-label="Open"
                        class="cds--list-box__menu-icon"
                        id="allergy-reactions-multiselect-test-allergy-1-toggle-button"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          focusable="false"
                          height="16"
                          preserveAspectRatio="xMidYMid meet"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                          />
                        </svg>
                      </button>
                    </div>
                    <ul
                      aria-labelledby="allergy-reactions-multiselect-test-allergy-1-label"
                      class="cds--list-box__menu"
                      id="allergy-reactions-multiselect-test-allergy-1__menu"
                      role="listbox"
                      style="position: fixed; left: 0px; top: 0px; visibility: visible; transform: translate(0px, 0px);"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="cds--sm:col-span-1 cds--md:col-span-1 cds--lg:col-span-1 cds--xlg:col-span-1 cds--css-grid-column"
          >
            <span
              class="cds--popover-container cds--popover--caret cds--popover--high-contrast cds--popover--top cds--tooltip cds--icon-tooltip"
            >
              <div
                class="cds--tooltip-trigger__wrapper"
              >
                <button
                  aria-label="Close Selected Item"
                  class="cds--btn cds--btn--lg cds--layout--size-lg cds--btn--primary cds--btn--icon-only"
                  data-testid="selected-item-close-button"
                  id="close-selected-item"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M17.4141 16L24 9.4141 22.5859 8 16 14.5859 9.4143 8 8 9.4141 14.5859 16 8 22.5859 9.4143 24 16 17.4141 22.5859 24 24 22.5859 17.4141 16z"
                    />
                  </svg>
                </button>
              </div>
              <span
                aria-hidden="true"
                class="cds--popover"
                id="tooltip-«r29»"
                role="tooltip"
              >
                <span
                  class="cds--popover-content cds--tooltip-content"
                >
                  Close
                </span>
                <span
                  class="cds--popover-caret"
                />
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
